drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -214,10 +249,94 @@ local function set_setpoint(setpoint)
       value = f_to_c(value)
     end
 
+    local cooling_setpoint = device.state_cache.main.thermostatCoolingSetpoint.coolingSetpoint
+    if cooling_setpoint and cooling_setpoint.unit == "F"then
+      cooling_setpoint = f_to_c(cooling_setpoint.value)
+    else
+      cooling_setpoint = cooling_setpoint.value
+    end
+    local heating_setpoint = device.state_cache.main.thermostatHeatingSetpoint.heatingSetpoint
+    if heating_setpoint and heating_setpoint.unit == "F"then
+      heating_setpoint = f_to_c(heating_setpoint.value)
+    else
+      heating_setpoint = heating_setpoint.value
+    end
+    print(string.format(
+      " heating setpoint (%s)  cooling setpoint (%s)",
+      heating_setpoint, cooling_setpoint
+     ))
+
+
+    local is_auto_capable = #device:get_endpoints(
+      clusters.Thermostat.ID,
+      {feature_bitmap = clusters.Thermostat.types.ThermostatFeature.AUTOMODE}
+    ) > 0
+    print("is_auto_capable", is_auto_capable)
+    --Check setpoint limits for the device
+    local setpoint_type = string.match(setpoint.NAME, "Heat") or "Cool"
+    -- local deadband = device:get_field(setpoint_limit_device_field.MIN_DEADBAND) or 0
+    local deadband = 1.3
```

Comment:
Should we assume a deadband of `2.5` when the field is failed to be read from the device (assuming the above line will be uncommented) since the attribute ([MinSetpointDeadBand](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/Thermostat.adoc#7-attributes)) is `2.5` by default and that is the maximum deadband value a device could have?